include/arch/arm/arch/64/mode/kernel/vspace.h

```diff
@@ -91,7 +89,11 @@ static inline exception_t performASIDPoolInvocation(asid_t asid, asid_pool_t *po
     asid_map_t asid_map;
     cap_page_upper_directory_cap_ptr_set_capPUDMappedASID(&cte->cap, asid);
     cap_page_upper_directory_cap_ptr_set_capPUDIsMapped(&cte->cap, 1);
+#ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
+    asid_map = asid_map_asid_map_vspace_new(cap_page_upper_directory_cap_get_capPUDBasePtr(cte->cap), 0, false);
```

Comment:
the change is mostly for arm64 hyp right? for the cores with 8-bit VMID. So the commit message could point out the VMID explicitly I guess.

Reply:
The change is for aarch64 hyp and also SMMUv2 and eventually kernel image clone. Longer term there is a plan that every architecture will have the asid_map_t feature if it wants to support the kernel image clone feature (a vspace can have a different kernel image to avoid shared cache timings).